jena-fuseki2/jena-fuseki-core/src/main/java/org/apache/jena/fuseki/servlets/ActionSPARQL.java

```diff
@@ -205,7 +206,11 @@ public static void parse(HttpAction action, StreamRDF dest, InputStream input, L
                 .lang(lang)
                 .base(base)
                 .parse(dest);
-        } 
+        } catch (RuntimeIOException ex) {
+            if ( ex.getCause() instanceof CharacterCodingException )
+                throw new RiotException("Character Coding Error: "+ex.getMessage());
```

Comment:
This is an error caught and handled as an HTTP servlet response and is a bad request - the client's mistake. It is not an error in the server to track down so I see this as normal processing from the server's point of view. The message is the information needed for the HTTP response.